File: Code coaches/Hard/Security/security.py
diagram = input()
diagram_splitted = diagram.split('$')

is_thief_on_left = True
thief_pos = None
guard_pos = None

for idx1, part in enumerate(diagram_splitted):
	if 'T' in part:
		for idx2, c in enumerate(part):
			if c.upper() == 'G':
				if not is_thief_on_left and guard_pos is not None:
					continue
				else:
					guard_pos = idx2
			elif c.upper() == 'T':
				thief_pos = idx2
	elif idx1 == 0:
		# if the field is not in the first array (left), then we know that it is in the right
		is_thief_on_left = False
		continue

is_left_robbed = (guard_pos is None or guard_pos < thief_pos) and is_thief_on_left
is_right_robbed = (guard_pos is None or guard_pos > thief_pos) and not is_thief_on_left

if is_left_robbed or is_right_robbed:
	print("ALARM")
else:
	print("quiet")

File: Code coaches/Hard/Security/security.cpp
#include <iostream>
#include <regex>

template <typename C, typename T>
bool contains(C &&c, T e)
{
	return find(begin(c), end(c), e) != end(c);
};

int main()
{
	std::string diagram;
	std::cin >> diagram;

	bool isThiefOnLeft = true;
	int thiefPos = 0;
	int guardPos = 0;

	std::regex reg("\\$");
	std::sregex_token_iterator iter(diagram.begin(), diagram.end(), reg, -1);
	std::sregex_token_iterator end;

	int idx1 = 0;
	std::vector<std::string> diagramSplitted(iter, end);
	for (auto part : diagramSplitted)
	{
		if (contains(part, 'T'))
		{
			int idx2 = 0;
			for (auto c : part)
			{
				if (toupper(c) == 'G')
				{
					if (!isThiefOnLeft && guardPos)
					{
						idx2++;
						continue;
					}
					else 
					{
						guardPos = idx2;
					}
				}
				else if (toupper(c) == 'T')
				{
					thiefPos = idx2;
				}

				idx2++;
			}
		}
		else if (idx1 == 0)
		{
			isThiefOnLeft = false;
			idx1++;
			continue;
		}		
		
		idx1++;
	}

	bool isLeftRobbed = (!guardPos || guardPos < thiefPos) && isThiefOnLeft;
	bool isRightRobbed = (!guardPos || guardPos > thiefPos) && !isThiefOnLeft;

	if (isLeftRobbed || isRightRobbed)
	{
		std::cout << "ALARM";
	}
	else 
	{
		std::cout << "quiet";
	}

	return 0;
}